src/PoolManager.sol

```diff
@@ -207,34 +218,71 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        key.hooks.beforeSwap(key, params, hookData);
-
-        uint256 feeForProtocol;
-        uint24 swapFee;
-        Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
-            Pool.SwapParams({
-                tickSpacing: key.tickSpacing,
-                zeroForOne: params.zeroForOne,
-                amountSpecified: params.amountSpecified,
-                sqrtPriceLimitX96: params.sqrtPriceLimitX96
-            })
+        // The hook's deltas are from the point of view of the hook. Positive: the hook took money, negative: the hook sent money to the pool
+        // TODO consider adding hookDeltaInUnspecified to beforeSwap hook too
```

Comment:
Can we add hookDeltaInSpecified to `afterSwap` hook as well so that either before or after can take fees on specified or unspecified tokens.

Reply:
@qiwu7  no thats not possible because the hookDeltaInSpecified has to alter the amountSpecified _before_ it gets swapped. If the amountSpecified to charge the user extra money then they will have to charge the user more money than `amountSpecified`. The UX we are looking to achieve with these deltas is that the user's action requested is not altered. For example in `swap`, the user states they want to send in (or receive) `amountSpecified` - and at the end of the swap no matter what happens that exact amount in/out will occur. What is altered is the actual amount swapped on the pool.

That means if you want to charge a fee on specified, you should do so in the `beforeSwap` hook.
- User specifies exactIn of `100e18`
- Hook takes `1e18` as a fee
- `99e18` is actually traded on the pool
- At the end the actual amount taken from the user is `100e18` which is exactly what they wanted their exactIn to be

If instead the `1e18` was added on top of the `100e18`, then the user would be charged `101e18` which is not what they specified in their call to `swap`.
Does that make sense?

Reply:
> Move this TODO to an issue? Unless you plan on addressing it in this PR.

Have added it as a comment in the issue which can remain open for now. Issue #463